bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureSmartLockThingHandler.java

```diff
@@ -0,0 +1,356 @@
+                        return;
+                    }
+
+                    ArrayList<SmartLock> list = new ArrayList<>();
+                    SmartLock smartLockJSON = new SmartLock();
+                    Variables variables = new Variables();
+                    Input input = new Input();
+
+                    variables.setDeviceLabel(deviceId);
+                    variables.setGiid(installationId.toString());
+                    input.setCode(pinCode);
+                    variables.setInput(input);
+                    smartLockJSON.setOperationName(operation);
+                    smartLockJSON.setVariables(variables);
+                    String query = "mutation " + operation
+                            + "($giid: String!, $deviceLabel: String!, $input: LockDoorInput!) {\n  " + operation
+                            + "(giid: $giid, deviceLabel: $deviceLabel, input: $input)\n}\n";
+                    smartLockJSON.setQuery(query);
+                    list.add(smartLockJSON);
+
+                    String queryQLSmartLockSetState = gson.toJson(list);
+                    logger.debug("Trying to set SmartLock state to {} with URL {} and data {}", operation, url,
+                            queryQLSmartLockSetState);
+
+                    int httpResultCode = session.sendCommand(url, queryQLSmartLockSetState, installationId);
+                    if (httpResultCode == HttpStatus.OK_200) {
+                        logger.debug("SmartLock status successfully changed!");
+                    } else {
+                        logger.warn("Failed to send command, HTTP result code {}", httpResultCode);
+                    }
+                } else {
+                    logger.warn("PIN code is not configured! It is mandatory to control SmartLock!");
+                }
+            }
+        }
+    }
+
+    private void handeAutoRelockResult(String url, String data, BigDecimal installationId, Command command) {
+        logger.debug("Trying to set Auto Relock {} with URL {} and data {}", command.toString(), url, data);
+        if (session != null) {
+            int httpResultCode = session.sendCommand(url, data, installationId);
+            if (httpResultCode == HttpStatus.OK_200) {
+                logger.debug("AutoRelock sucessfully changed to {}", command.toString());
+            } else {
+                logger.warn("Failed to send command, HTTP result code {}", httpResultCode);
+            }
+        }
+    }
+
+    private void handleAutoRelock(Command command) {
+        String deviceId = config.getDeviceId();
+        if (session != null) {
+            VerisureSmartLocks smartLock = (VerisureSmartLocks) session.getVerisureThing(deviceId);
+            if (smartLock != null) {
+                BigDecimal installationId = smartLock.getSiteId();
+                String csrf = session.getCsrfToken(installationId);
+                StringBuilder sb = new StringBuilder(deviceId);
+                sb.insert(4, "+");
```

Comment:
Why insert at index 4? Is this supposed to append a '+' to deviceId?

Reply:
Device ID is a 9 character field that most often is used like this in API calls:

```
deviceLabel":"34A58JU5"
```

But this particular call demands device ID to be:

```
enabledDoorLocks=34A5+8JU5
```